system/include/emscripten/wasm_worker.h

```diff
@@ -218,7 +223,7 @@ int emscripten_semaphore_try_acquire(emscripten_semaphore_t *sem __attribute__((
 // acquired. If you use this API in Worker, you cannot run an infinite loop.
 void emscripten_semaphore_async_acquire(emscripten_semaphore_t *sem __attribute__((nonnull)),
                                         int num,
-                                        void (*asyncWaitFinished)(volatile void *address, uint32_t idx, ATOMICS_WAIT_RESULT_T result, void *userData) __attribute__((nonnull)),
+                                        emscripten_async_wait_callback_t asyncWaitFinished __attribute__((nonnull)),
```

Comment:
This is volatile so shouldn't it be the other typedef, `emscripten_async_lock_acquire_callback_t`?